### ISSUE TITLE
AV1 Codec support

### ISSUE THREAD

**Issue by gareth-anthony-hulse** [alvr-org/ALVR#1501]
Would like to have the AV1 codec as an option for those of use who can't achieve a decent bit rate. I am using the D-Link DWA-X1850 with the [rtl8852au](https://github.com/lwfinger/rtl8852au) Linux kernel driver, as an access point using the 802.11ac standard. Current setup can achieve ~16Mbps average at best, sometimes with sudden dips into single digits. Using HEVC makes it tolerable, but the blurriness is definitely noticeable when looking at objects from a distance or down iron sights without fibre optics.

**Hardware:** https://linux-hardware.org/?probe=382901fcd7

**Comment by Vixea** [comment]
That would have to be done with software decoding the quest or really the XR2 is simply not powerful enough to make this a good experience.
Anyway my guess is that you have ALVR mis-configured.

**Comment by gareth-anthony-hulse** [comment]
My ALVR config: [session.zip](https://github.com/alvr-org/ALVR/files/10947655/session.zip)

**Comment by Vixea** [comment]
could you also give me a screenshot of the video tab in settings mostly interested in the area that says bitrate

**Comment by Vixea** [comment]
anyway try changing the foveated encoding ratio to 2 and 3(top one 2, bottom one 3

**Comment by gareth-anthony-hulse** [comment]
This what is looks like at the moment: ![Screenshot_20230311_051558](https://user-images.githubusercontent.com/107568353/224466382-af9660ca-a8da-47ce-ae4a-33fe5d16327a.png)

I'll try editing the foveated encoding.

**Comment by Vixea** [comment]
how did you come up with the number 6 could you try 8 and see if that increases it?

**Comment by gareth-anthony-hulse** [comment]
Changed 'networkLaytencyMs' to 8, Horizontal compression ratio to 2 and vertical compression radio to 3. Ran Blade & Sorcery for couple of minutes and here are the results: ![Screenshot_20230311_055013](https://user-images.githubusercontent.com/107568353/224467527-bc308f1c-631d-40c8-ab1f-67803580bcfd.png)

**Comment by Vixea** [comment]
try changing it saturation instead

**Comment by gareth-anthony-hulse** [comment]
Using 'adaptiveWithSaturation' with saturation multiplier of 0.95, while playing H3VR (Blade & Sorcery was getting stuck at 84% for some reason): ![Screenshot_20230311_062742](https://user-images.githubusercontent.com/107568353/224468994-0f583e49-e61f-4c4a-9442-2f377a6605c8.png)

**Comment by gareth-anthony-hulse** [comment]
Oh and I am using ALVR version v20.0.0-dev07+nightly.2023.03.09, with SteamVR 1.25.5.

**Comment by Vixea** [comment]
have you tried restarting your router :D

**Comment by Vixea** [comment]
also for the blurriness did you not set the resolution to be 100%?

**Comment by gareth-anthony-hulse** [comment]
The Quest 2 isn't directly connected to the ASUS GT-AX11000 router, but I'll give it a try. The headset is connected to the D-Link DWA-X1850, which is setup as a WiFi 5 access point that is plugged into the PC via a USB 3.0 socket and the PC is connected to the router via WiFi 6.

**Comment by Vixea** [comment]
hmm, wifi hotspot support isn't exactly the best in windows could you try the router instead?

**Comment by gareth-anthony-hulse** [comment]
I am using Linux - see hardware link in first post. Haven't tried the router for a while because the latency was constantly high. I'll give it another go.

**Comment by gareth-anthony-hulse** [comment]
Result when using the ASUS GT-AX11000 router in H3VR - take note of the Mbps and client FPS: ![Screenshot_20230311_074001](https://user-images.githubusercontent.com/107568353/224471962-a043f198-6cbf-489a-8f4e-f35a70d2176e.png)

**Comment by gareth-anthony-hulse** [comment]
Back on the subject of the AV1 codec, while the current gen VR headsets don't support it very will, future gens hopefully will.

**Comment by Vixea** [comment]
and I'm sure we'll add it when its ready as right now this is a won't fix

**Comment by wsippel** [comment]
Now that the Quest 3 is out and supports AV1, with Meta defaulting to AV1 for Air Link as far as I understand, I guess this could be reopened?

**Comment by phobos2077** [comment]
Please reopen this. AV1 is a game changer on Q3 based on what VD users report.

**Comment by zarik5** [comment]
I will work on AV1 support for Windows/NVENC soon.

**Comment by wsippel** [comment]
Very nice! I already have patches to enable AV1 encoding on Linux using VA-API, I just couldn't figure out how to decode the stream on the client. So Linux support could follow quickly once the Windows patches land.

**Comment by phobos2077** [comment]
Plz reopen, AV1 is still not supported on Windows.

**Comment by Vixea** [comment]
It is in fact partially supported on windows (you need to have an AMD GPU), we know Nvidia on windows doesn't support it, it's not critical, and it was going to be closed soon anyway by the bot so. No

**Comment by zarik5** [comment]
@Vixea It's not supported on Windows/Nvidia, it's a real issue.

**Comment by Vixea** [comment]
You need to be sleeping :P

**Comment by phobos2077** [comment]
I believe Windows/NVidia is still not implemented.

**Comment by zmerp** [comment]
Since windows/nvidia av1 encoder has been implemented, this can be closed
